### PR TITLE
Moved run loop and added phyx init

### PR DESCRIPTION
**Pull request by JPlosz**
Main application loop taken out of renderer.run() and placed in engine.run().
Added enough physx code to initialize it's main variables.  No added actors to the scene or rendering implemented.
Print msg to the console when physx is done initialization.

### CODE REVIEW

File: src/Renderer.h [hunk 1]
```diff
@@ -30,6 +36,14 @@ class Renderer
 		const unsigned int height = 800;
 		const unsigned int width = 800;
 
+		enum Scene
+		{
+			Test = 0,
+			PhysX = 1,
+		};
+
+		int sceneSelect;
```
Comment (manjot-b) [inline src/Renderer.h:45]: Just wondering what are these used for? I saw that you toggle `sceneSelect` when space is pressed but I don't see it being used for anything.
Maybe add a short comment if it is needed or remove it if it was just for some testing.

File: src/Simulate.cpp [hunk 2]
```diff
@@ -1,13 +1,86 @@
...
+	gPhysics->release();
+	if (gPvd)
+	{
+		PxPvdTransport* transport = gPvd->getTransport();
+		gPvd->release();	gPvd = NULL;
+		transport->release();
+	}
+	gFoundation->release();
+
+	printf("SnippetHelloWorld done.\n");
```
Comment (manjot-b) [inline src/Simulate.cpp]: Wanna remove this line?

`cleanupPhysics()` isn't called anywhere but I can handle that tomorrow when I do a little bit of reorganizing of the these classes. I will clean up PhysX when cleaning up GLFW.

**Review by manjot-b — CHANGES_REQUESTED** [review]

File: src/Renderer.h (hunk 1 above)
Comment (JPlosz) [reply src/Renderer.h:45]: I had a way to start the program in the original scene labelled "Test" to see everything working as it did before but wanted a quick way to switch to the physX scene.  Might be useful when we have Menu, inGame, Pause windows to run through

Comment (manjot-b) [reply src/Renderer.h:45]: Ok we can just leave it for now then.

**Pull request merged**